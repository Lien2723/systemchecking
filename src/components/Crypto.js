/* eslint-disable jsx-a11y/control-has-associated-label */
import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import '../App.css';

export default function Crypto({
  image, id, currentPrice, name, symbol,
}) {
  return (
    <>
      <li className="list-items">
        <Link to={`/detail/${id}`} id={id} className="nav-link">
          <div className="main-Container">
            <img className="coin-image" src={image} alt={image} />
            <div className="crypto-details">
              <h2 className="main-page-each-li-detail">
                <span className="detail-feature-tag">Name:</span>
              &nbsp;
                {name}
              </h2>
              <h2 className="main-page-each-li-detail">
                <span className="detail-feature-tag">Symbol:</span>
              &nbsp;
                {symbol}
              </h2>
              <h2 className="main-page-each-li-detail">
                <span className="detail-feature-tag">Price:</span>
              &nbsp;
                {currentPrice}
              &nbsp;
                <span className="USD" style={{ color: '#0aa756' }}>$</span>
              </h2>
            </div>
            <Link to={`/detail/${id}`} id={id}>
              <div className="detail_btn">
                <i className="fa-solid fa-circle-arrow-right ml-auto detail-btn" />
              </div>
            </Link>
          </div>
        </Link>
      </li>
    </>
  );
}

Crypto.propTypes = {
  image: PropTypes.string.isRequired,
  id: PropTypes.string.isRequired,
  currentPrice: PropTypes.number.isRequired,
  name: PropTypes.string.isRequired,
  symbol: PropTypes.string.isRequired,
};
