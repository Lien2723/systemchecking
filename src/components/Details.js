/* eslint-disable react/prop-types */
import React from 'react';

export default function Details({
  image,
  currentPrice,
  name,
  symbol,
  marketCap,
  marketCapRank,
  totalVolume,
  high24h,
  low24h,
  priceChange24h,
  marketCapChange24h,
  totalSupply,
}) {
  // const { crypto } = props;

  return (
    <div data-testid="Details-1" className="detail-page-all">
      <div className="coin-img-sec">
        <img className="coin-img" src={image} alt={image} />
      </div>
      <div className="info-text">
        <p>Crypto Coin Info:</p>
      </div>
      <ul className="detail-ul-">
        <li className="detail-">
          <span className="coin-name">Name:</span>
          <span className="coin-res">
            {name}
            {' '}
            <i className="fa-solid fa-circle-arrow-right" />
          </span>
        </li>
        <li className="detail-">
          <span className="coin-name">Symbol: </span>
          <span className="coin-res">
            {symbol}
            {' '}
            <i className="fa-solid fa-circle-arrow-right pl-2" />
          </span>
        </li>
        <li className="detail-">
          <span className="coin-name">Price: </span>
          <span className="coin-res">
            {currentPrice}
            {' '}
            $
            <i className="fa-solid fa-circle-arrow-right pl-2" />
          </span>
        </li>
        <li className="detail-">
          <span className="coin-name">Market Cap:</span>
          <span className="coin-res">
            {marketCap}
            <i className="fa-solid fa-circle-arrow-right pl-2" />
          </span>
        </li>
        <li className="detail-">
          <span className="coin-name">Market Cap Rank:</span>
          <span className="coin-res">
            {marketCapRank}
            <i className="fa-solid fa-circle-arrow-right pl-2" />
          </span>
        </li>
        <li className="detail-">
          <span className="coin-name">Total Volume:</span>
          <span className="coin-res">
            {totalVolume}
            <i className="fa-solid fa-circle-arrow-right pl-2" />
          </span>
        </li>
        <li className="detail-">
          <span className="coin-name">Highest 24h:</span>
          <span className="coin-res">
            {high24h}
            <i className="fa-solid fa-circle-arrow-right pl-2" />
          </span>
        </li>
        <li className="detail-each-li">
          <span className="coin-name">Lowest 24h</span>
          <span className="coin-res">
            {low24h}
            <i className="fa-solid fa-circle-arrow-right pl-2" />
          </span>
        </li>
        <li className="detail-">
          <span className="coin-name">Price Change 24h:</span>
          <span className="coin-res">
            {priceChange24h}
            <i className="fa-solid fa-circle-arrow-right pl-2" />
          </span>
        </li>
        <li className="detail-">
          <span className="coin-name">Market Cap Change</span>
          <span className="coin-res">
            {marketCapChange24h}
            <i className="fa-solid fa-circle-arrow-right pl-2" />
          </span>
        </li>
        <li className="detail-">
          <span className="coin-name">Total Supply:</span>
          <span className="coin-res">
            {totalSupply}
            <i className="fa-solid fa-circle-arrow-right pl-2" />
          </span>
        </li>
      </ul>
    </div>
  );
}
