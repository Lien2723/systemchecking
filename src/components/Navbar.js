const Navbar = () => (
  <nav>
    <div className="Nav-section">
      <h1 className="header">Cryptocurrency</h1>
    </div>
  </nav>
);

export default Navbar;
