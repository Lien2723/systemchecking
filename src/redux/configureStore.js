import { combineReducers, createStore, applyMiddleware } from 'redux';
import thunk from 'redux-thunk';
import { cryptosReducer } from './cryptos/Cryptos';

const reducer = combineReducers({
  cryptos: cryptosReducer,
});

export const store = createStore(
  reducer,
  applyMiddleware(thunk),
);

export default store;
