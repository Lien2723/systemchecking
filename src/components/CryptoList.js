/* eslint-disable react/prop-types */
import React from 'react';
import Crypto from './Crypto';

export const CryptoList = (props) => {
  const { cryptos } = props;

  return (
    <ul className="crypto-ul">
      {cryptos.map((crypto) => (
        <Crypto
          key={crypto.id}
          image={crypto.image}
          id={crypto.id}
          currentPrice={crypto.currentPrice}
          name={crypto.name}
          symbol={crypto.symbol}
        />
      ))}
    </ul>
  );
};

export default CryptoList;
