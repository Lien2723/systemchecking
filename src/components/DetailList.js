/* eslint-disable react/prop-types */
import React from 'react';
import { useParams } from 'react-router-dom';
import Details from './Details';

export const DetailsList = (props) => {
  const { cryptos } = props;
  const params = useParams();

  const specificDetailToken = [...cryptos.filter((crypto) => crypto.id === params.id)];

  return (
    <ul className="details-ul">
      {specificDetailToken
        && specificDetailToken.map((crypto) => (
          <Details
            key={crypto.id}
            image={crypto.image}
            currentPrice={crypto.currentPrice}
            name={crypto.name}
            symbol={crypto.symbol}
            marketCap={crypto.symbol}
            marketCapRank={crypto.marketCap}
            totalVolume={crypto.totalVolume}
            high24h={crypto.high24h}
            low24h={crypto.low24h}
            priceChange24h={crypto.priceChange24h}
            marketCapChange24h={crypto.marketCapChange24h}
            marketCapChangePercentage24h={crypto.marketCapChangePercentage24h}
            totalSupply={crypto.totalSupply}
          />
        ))}
    </ul>
  );
};

export default DetailsList;
