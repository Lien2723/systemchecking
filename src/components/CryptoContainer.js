/* eslint-disable jsx-a11y/control-has-associated-label */
import React, { useEffect, useState } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import { CryptoList } from './CryptoList';
import { LoadCryptos } from '../redux/cryptos/Cryptos';

export const CryptoContainer = () => {
  const coin = useSelector((state) => state.cryptos.cryptos);
  const dispatch = useDispatch();

  const [search, setSearch] = useState('');

  useEffect(() => {
    dispatch(LoadCryptos());
  }, [dispatch]);

  return (
    <div className="all-content">
      <div className="main-container">
        <div className="seach-section">
          <div className="searchBox">
            <input
              type="text"
              className="searchInput"
              placeholder="Search..."
              onChange={(event) => setSearch(event.target.value)}
            />
            <button type="button" className="searchButton" onChange={(e) => setSearch(e.target.value)}><i className="fas fa-search" /></button>
          </div>
        </div>
      </div>
      <div className="crypto-items">
        <CryptoList cryptos={coin.filter((crypto) => (search.toLowerCase() === '' ? crypto : crypto.name.toLowerCase().includes(search.toLowerCase())))} />

      </div>
    </div>
  );
};

export default { CryptoContainer };
