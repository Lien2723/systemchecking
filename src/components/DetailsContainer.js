import React, { useEffect } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import { DetailsList } from './DetailList';
import { LoadCryptos } from '../redux/cryptos/Cryptos';

export const DetailsContainer = () => {
  const cryptos = useSelector((state) => state.cryptos.cryptos);
  const dispatch = useDispatch();

  useEffect(() => {
    dispatch(LoadCryptos());
  }, [dispatch]);

  return (
    <div className="coin-items">
      <div className="main-container-details">
        <a className="back-pointer" href="/">
          <i className="fa-solid fa-left-long" />
        </a>
        <h2 className="details-sec">Details</h2>
      </div>
      <DetailsList cryptos={cryptos} />
    </div>
  );
};

export default { DetailsContainer };
