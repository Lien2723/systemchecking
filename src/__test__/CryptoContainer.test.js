import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import '@testing-library/jest-dom';
import { BrowserRouter } from 'react-router-dom';
import { CryptoContainer } from '../components/CryptoContainer';

import { store } from '../redux/configureStore';

it('Renders Crypto Home page', () => {
  const tree = renderer.create(
    <Provider store={store}>
      <BrowserRouter>
        <CryptoContainer />
      </BrowserRouter>
    </Provider>,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
