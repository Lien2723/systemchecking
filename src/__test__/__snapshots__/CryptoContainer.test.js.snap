// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders Crypto Home page 1`] = `
<div
  className="all-content"
>
  <div
    className="main-container"
  >
    <div
      className="seach-section"
    >
      <div
        className="searchBox"
      >
        <input
          className="searchInput"
          onChange={[Function]}
          placeholder="Search..."
          type="text"
        />
        <button
          className="searchButton"
          onChange={[Function]}
          type="button"
        >
          <i
            className="fas fa-search"
          />
        </button>
      </div>
    </div>
  </div>
  <div
    className="crypto-items"
  >
    <ul
      className="crypto-ul"
    />
  </div>
</div>
`;
